include/transform.h

```diff
@@ -0,0 +1,127 @@
+        Default: [1e-4, -1e-4, 1e-6, 1e-6, 1e-8, -1e-8, 1e-10, -1e-10, 1e-12, -1e-12, 1e-14, -1e-14]
+        Verification will test all values in order until at least one passes.
+    :param kwargs: dict; args to pass to np.isclose when comparing derivative to finite difference e.g.
+        atol, rtol
+    :return:
+    """
+    if parameter.transform is None:
+        return None
+    if parameter.transform.derivative is None:
+        if not return_finite_diff:
+            return None
+        else:
+            value = parameter.get_value()
+            if dx is None:
+                if dx_ratios is None:
+                    raise ValueError('Finite difference derivative for parameter {} must specify one of'
+                                     'dx or dx_ratios'.format(parameter))
+                dx = dx_ratios[0] * value
+            return (parameter.transform.forward(value + dx) - parameter.get_value_transformed()) / dx
+    value = parameter.get_value()
+    derivative = parameter.transform.derivative(value)
+    if verify:
+        value_transformed = parameter.get_value_transformed()
+        # Skip testing finite differencing if the derivative is very large
+        # This might happen e.g. near the limits of the transformation
+        # TODO: Check if better finite differencing is possible for large values
+        if verify_derivative_abs_max is None:
+            verify_derivative_abs_max = 1e8
+        is_close = np.abs(derivative) > verify_derivative_abs_max
+        if not is_close:
+            if dx_ratios is None:
+                dx_ratios = [1e-4, 1e-6, 1e-8, 1e-10, 1e-12, 1e-14]
+            for ratio in dx_ratios:
+                dx = value * ratio
+                fin_diff = (parameter.transform.forward(value + dx) - value_transformed) / dx
+                if not np.isfinite(fin_diff):
+                    fin_diff = -(parameter.transform.forward(value - dx) - value_transformed) / dx
+                is_close = np.isclose(derivative, fin_diff, **kwargs)
+                if is_close:
+                    break
+        if not is_close:
+            raise RuntimeError(
+                'Param {}(t({})={}) derivative={:.8e} != last '
+                'finite diff.={:8e} with dx={} dx_abs_max={}'.format(
+                    parameter.name, value, value_transformed, derivative, fin_diff, dx,
+                    verify_derivative_abs_max
+                ))
+    return derivative
+*/
+template <class T>
+class Transform {
+public:
+    virtual std::string description() const = 0;
+    virtual std::string str() const = 0;
+
+    virtual T derivative(T x) const = 0;
+    virtual T forward(T x) const = 0;
+    virtual T reverse(T x) const = 0;
+    // TODO: Find a solution for a destructor that compiles without warnings
+    // May not be possible; see http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/p1077r0.html
+    // virtual ~Transform() = default; // or delete; neither works
```

Comment:
If it's virtual, it can't be trivial; if it's not trivial, you can't declare e.g. `static const UnitTransform<T> transform_none = UnitTransform<T>()`. This isn't a problem for `limits_maximal` because the Limits class has no base class, abstract or otherwise.